### ISSUE TITLE
Regression: Relative dates should be implicitly anchored around 'now'

### ISSUE THREAD

**Issue by DominikRusso** [GothenburgBitFactory/taskwarrior#2390]
"Correct" behavior in `2.5.1` with minimal taskrc:
```
> task --version
2.5.1
> task add test due:7d
Created task 1.
> task list due.before:7d

ID Age  Due        Description Urg
 1 1min 2021-01-25 test         5.6

1 task
> task list due.after:7d
No matches.
```

"Incorrect" behavior in `2.5.3` with minimal taskrc:
```
> task --version
2.5.3
> task add test due:7d
Created task 1.
>  task list due.before:7d
No matches.
> task list due.after:7d

ID Age  Due        Description Urg
 1 1min 2021-01-25 test         5.6

1 task
```
but also
```
> task list due.after:10y

ID Age  Due        Description Urg
 1 1min 2021-01-25 test         5.6

1 task
```

Prepending `now+` to the relative dates yields expected behavior.

### `task diag`
```
task 2.5.3
   Platform: Linux

Compiler
    Version: 10.2.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++17

Build Features
      Built: Jan 18 2021 02:04:50
     Commit: 2f47226f9
      CMake: 3.19.3
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.7.0
 Build type: release

Configuration
       File: /home/xxx/.config/task/taskrc (found), 181 bytes, mode 100644
       Data: /home/xxx/.local/share/task (found), dir, mode 40755
     TASKRC: /home/xxx/.config/task/taskrc
    Locking: Enabled
         GC: Enabled
    $VISUAL: nvim
     Server:
         CA: -
Certificate: -
        Key: -
      Trust: strict
    Ciphers: NORMAL
      Creds:

Hooks
     System: Enabled
   Location: /home/xxx/.local/share/task/hooks
             (-none-)

Tests
   Terminal: 120x30
       Dups: Scanned 3 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 3 tasks for broken references:
             No broken references found

```

**Comment by tbabej** [comment]
Reproduced on 2.6.0. There seems to be a problem with conversion of durations into dates, as evidenced with `task calc`:

```
[warrior@tw-2.6.0 ~]$ task calc now \< P7D
false
[warrior@tw-2.6.0 ~]$ task calc now \< now + P7D
true
```

**Comment by tbabej** [comment]
This is already fixed and shipping as part of 2.6.0, but for people stuck on 2.5.2 / 2.5.3, the workaround here is to use datetime expressions `due.before:now+7d` instead of `due.before:7d`.
